### PR TITLE
Rd 5977 validate input types are matching dsl version

### PR DESCRIPTION
**Pull request by kaplanyaniv**

### CODE REVIEW

File: cfy_lint/yamllint_ext/rules/inputs.py [hunk 1]
```diff
@@ -82,6 +93,11 @@ def validate_inputs(input_obj, line):
             if isinstance(input_obj.default, list):
                 message += 'The correct type could be "list".'
         yield LintProblem(line, None, message)
+    elif input_obj.input_type not in INPUTS_BY_DSL.get(dsl, []):
+        yield LintProblem(line, None, 'input "{}" of type "{}" doesn\'t work '
```
Comment (EarthmanT) [inline cfy_lint/yamllint_ext/rules/inputs.py:97]: Let's improve this problem message. Like `"Input of type {} is not supported by DSL {}."`.

File: cfy_lint/yamllint_ext/rules/inputs.py [hunk 2]
```diff
@@ -28,6 +28,15 @@
 TYPE = 'token'
 CONF = {'allowed-values': list(VALUES), 'check-keys': bool}
 DEFAULT = {'allowed-values': ['true', 'false'], 'check-keys': True}
+INPUTS_BY_DSL = {
+    'cloudify_dsl_1_3': ['string', 'integer', 'float', 'boolean', 'list',
+                         'dict', 'regex', 'textarea'],
```
Comment (EarthmanT) [inline cfy_lint/yamllint_ext/rules/inputs.py:33]: Yeah we support `textarea`?

File: cfy_lint/yamllint_ext/rules/inputs.py [hunk 3]
```diff
@@ -28,6 +28,15 @@
...
+    'cloudify_dsl_1_4': ['string', 'integer', 'float', 'boolean', 'node_id',
+                         'dict', 'regex', 'textarea', 'blueprint_id', 'list',
+                         'deployment_id', 'capability_value', 'scaling_group',
+                         'node_type', 'node_instance', 'secret_key',
+                         'node_ids', 'node_template', 'node_instance_ids',
+                         'deployment_ids', 'blueprint_ids']}
```
Comment (EarthmanT) [inline cfy_lint/yamllint_ext/rules/inputs.py:39]: I would find a way for 1_4 to be:
```
DSL_14 = DSL_13.extend([blueprint_id, ...])
```

File: cfy_lint/yamllint_ext/rules/inputs.py (hunk 2 above)
Comment (kaplanyaniv) [reply cfy_lint/yamllint_ext/rules/inputs.py]: I asked mateusz and he told me that we do, I never saw it being used but apparently we do

**Review by EarthmanT — CHANGES_REQUESTED** [review]

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**